src/test/javascript/spec/component/statistics/statistics-graph.component.spec.ts

```diff
@@ -0,0 +1,125 @@
+import * as chai from 'chai';
+import * as sinonChai from 'sinon-chai';
+import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { ArtemisTestModule } from '../../test.module';
+import { MockSyncStorage } from '../../helpers/mocks/service/mock-sync-storage.service';
+import { LocalStorageService, SessionStorageService } from 'ngx-webstorage';
+import { MockComponent, MockPipe } from 'ng-mocks';
+import { RouterTestingModule } from '@angular/router/testing';
+import { TranslatePipe, TranslateService } from '@ngx-translate/core';
+import { MomentModule } from 'ngx-moment';
+import { StatisticsService } from 'app/admin/statistics/statistics.service';
+import { StatisticsGraphComponent } from 'app/admin/statistics/statistics-graph.component';
+import { Graphs, SpanType } from 'app/entities/statistics.model';
+import { MockTranslateService } from '../../helpers/mocks/service/mock-translate.service';
+import { of } from 'rxjs/internal/observable/of';
+import * as moment from 'moment';
+import { SimpleChange } from '@angular/core';
+import { ChartsModule } from 'ng2-charts';
+import { HttpTestingController } from '@angular/common/http/testing';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+describe('StatisticsGraphComponent', () => {
+    let fixture: ComponentFixture<StatisticsGraphComponent>;
+    let component: StatisticsGraphComponent;
+    let service: StatisticsService;
+    let httpMock: HttpTestingController;
+
+    beforeEach(fakeAsync(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, RouterTestingModule.withRoutes([]), MomentModule, ChartsModule],
+            declarations: [StatisticsGraphComponent, MockComponent(StatisticsGraphComponent), MockPipe(TranslatePipe)],
+            providers: [
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+                { provide: TranslateService, useClass: MockTranslateService },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(StatisticsGraphComponent);
+                component = fixture.componentInstance;
+                service = TestBed.inject(StatisticsService);
+                httpMock = TestBed.get(HttpTestingController);
+            });
+    }));
+
+    afterEach(fakeAsync(() => {
+        jest.clearAllMocks();
```

Comment:
This should probably also call `sinon.restore()`?

Reply:
I cannot find a usage of sinon in my test class, so i thought say i dont need a sinon.restore()?

Reply:
We're using spies to "override" methods, `sinon.restore` is meant to "clean" that up.

Reply:
Yes but spy and spyOn is a Jasmine command, this has nothing to do with sinon, right?

Reply:
:eyes: color me surprised, I thought we used sinon here. Forget it then.